### PR TITLE
chore: replace session wide fixtures with scope

### PR DESCRIPTION
**Pull request by kostasrim**
* replace all session wide df fixtures with scope

### CODE REVIEW

File: tests/dragonfly/conftest.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def parse_args(args: List[str]) -> Dict[str, Union[str, None]]:
     return args_dict
 
 
-@pytest.fixture(scope="session", params=[{}])
+@pytest.fixture(scope="function", params=[{}])
```
Comment (adiholden) [inline tests/dragonfly/conftest.py:80]: you should change this to function as well

@pytest.fixture(scope="class")
def connection(df_server: DflyInstance):

**Comment by adiholden** [comment]
also there are some tests in snapshot_test.py to fix as they use the same dbfilename for few test cases and this will fail as you run the next test case it will try to load the rdb file and executing the first command will result in error as dragonfly is loading
we need to fix this tests to use different dbfilename for each test case

**Comment by kostasrim** [comment]
> also there are some tests in snapshot_test.py to fix as they use the same dbfilename for few test cases and this will fail as you run the next test case it will try to load the rdb file and executing the first command will result in error as dragonfly is loading we need to fix this tests to use different dbfilename for each test case

I remember you saying this but I wanted them to fail here since it will take a little less time than running them on my machine while multitasking :rofl:

**Comment by kostasrim** [comment]
https://github.com/dragonflydb/dragonfly/actions/runs/9744559739 regression tests

**Review by adiholden — APPROVED** [review]

**Pull request merged**